LBH.AdultSocialCare.Api/V1/UseCase/CarePackages/Concrete/GetServiceUserPackagesUseCase.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            _carePackageGateway = carePackageGateway;
+            _serviceUserGateway = serviceUserGateway;
+        }
+
+        public async Task<ServiceUserPackagesViewResponse> ExecuteAsync(Guid serviceUserId)
+        {
+            // Check service user exists
+            var serviceUser = await _serviceUserGateway.GetUsingIdAsync(serviceUserId).EnsureExistsAsync($"Service user with id {serviceUserId} not found");
+
+            const PackageFields fields = PackageFields.Details | PackageFields.Reclaims;
+            var userPackages = await _carePackageGateway.GetServiceUserPackagesAsync(serviceUserId, fields);
+            var response = new ServiceUserPackagesViewResponse
+            {
+                ServiceUser = serviceUser.ToBasicDomain().ToResponse(),
+                Packages = new List<ServiceUserPackageViewItemResponse>()
+            };
+
+            var packagesResponse = new List<ServiceUserPackageViewItemResponse>();
+
+            foreach (var carePackage in userPackages)
+            {
+                var packageResponse = new ServiceUserPackageViewItemResponse
+                {
+                    PackageId = carePackage.Id,
+                    PackageStatus = carePackage.Status.GetDisplayName(),
+                    PackageType = carePackage.PackageType.GetDisplayName(),
+                    DateAssigned = carePackage.DateAssigned,
+                    GrossTotal = 0,
+                    NetTotal = 0,
+                    PackageItems = new List<CarePackageCostItemResponse>()
+                };
+
+                var coreCost = carePackage.Details
+                    .SingleOrDefault(d => d.Type is PackageDetailType.CoreCost);
+
+                var additionalNeeds = carePackage.Details
+                    .Where(d => d.Type == PackageDetailType.AdditionalNeed).ToList();
+
+                packageResponse.PackageItems = CollectPackageItems(carePackage, coreCost, additionalNeeds, carePackage.Reclaims.ToList());
+                packagesResponse.Add(packageResponse);
+            }
+
+            response.Packages = packagesResponse;
+
+            return response;
+        }
+
+        private static IEnumerable<CarePackageCostItemResponse> CollectPackageItems(CarePackage package, CarePackageDetail coreCost, IReadOnlyCollection<CarePackageDetail> additionalNeeds, IReadOnlyCollection<CarePackageReclaim> reclaims)
+        {
+            // Add core cost
+            var carePackageCostItem = new List<CarePackageCostItemResponse>
+            {
+                new CarePackageCostItemResponse
+                {
+                    Id = package.Id,
+                    Name = package.PackageType.GetDisplayName(),
+                    Type = package.PackageType.GetDisplayName(),
+                    CollectedBy = ClaimCollector.Hackney.GetDisplayName(),
+                    Status = DateTimeOffset.Now.IsInRange(coreCost.StartDate,
```

Comment:
No need to auto calculate for the end. The user is responsible for ending manually.

Reply:
@brkzkn opinion

Reply:
I got your point Duncan. 

CoreElement status should be the same as Package status.
For ANPs, we should keep status column. Status might be `Active, End, Cancelled`. Active, End status comes from Package status. When the user remove ANPs, status will be `Cancelled`.